crud/select/executor.lua

```diff
@@ -53,11 +55,18 @@ function executor.execute(space, index, filter_func, opts)
         if value == nil then
             value = opts.after_tuple
         else
-            local cmp_operator = select_comparators.get_cmp_operator(opts.tarantool_iter)
-            local scan_comparator = select_comparators.gen_tuples_comparator(cmp_operator, index.parts)
-            local after_tuple_key = utils.extract_key(opts.after_tuple, index.parts)
-            if scan_comparator(after_tuple_key, opts.scan_value) then
-                value = after_tuple_key
+            if has_keydef then
+                local key_def = keydef_lib.new(index.parts)
+                if key_def:compare_with_key(opts.after_tuple, opts.scan_value) < 0 then
+                    value = key_def:extract_key(opts.after_tuple)
```

Comment:
Seems that this `if`s with `key_def:extract_key` or `utils.extract_key` can be moved to a separated function that knows if key def can be used or not.

Reply:
I left a comment for this topic https://github.com/tarantool/crud/commit/eccaff5123ba729574afbdf4583ee23bc6e5109b#r52462548